### PR TITLE
Show registration link at idp discovery identifier page

### PR DESCRIPTION
**Pull request by haishengwu-okta**
- support RaaS when idp discovery is enabled
- video: https://okta.box.com/s/olz3hsws3rzgqvdj8ty9fp18kxm040t5
- OKTA-154930

@yuliu-okta @mauriciocastillosilva-okta 
@santhoshbalakrishnan @uvartak-okta 
@yingli-okta

### CODE REVIEW

File: test/unit/spec/IDPDiscovery_spec.js [hunk 1]
```diff
@@ -1036,5 +1046,47 @@ function (_, $, Q, OktaAuth, LoginUtil, Okta, Util, AuthContainer, IDPDiscoveryF
...
+      });
+      itp('does not show the registration button if features.registration is false', function () {
+        var registration =  {
+        };
+        return setupRegistrationButton(null, registration).then(function (test) {
+          expect(test.form.registrationContainer().length).toBe(0);
+        });
+      });
+      itp('show the registration button if settings.registration.enable is true', function () {
+        var registration =  {
```
Comment (santhoshbalakrishnan) [inline test/unit/spec/IDPDiscovery_spec.js]: Is this a valid usecase? Curious if this can happen actually.

**Comment by yiyangli-okta** [comment]
🚢

File: test/unit/spec/IDPDiscovery_spec.js [hunk 2]
```diff
@@ -1036,5 +1046,47 @@ function (_, $, Q, OktaAuth, LoginUtil, Okta, Util, AuthContainer, IDPDiscoveryF
...
+    Expect.describe('Registration Flow', function () {
+      itp('does not show the registration button if features.registration is not set', function () {
+        return setup().then(function (test) {
+          expect(test.form.registrationContainer().length).toBe(0);
+        });
+      });
+      itp('does not show the registration button if features.registration is false', function () {
+        var registration =  {
+        };
+        return setupRegistrationButton(null, registration).then(function (test) {
```
Comment (mauriciocastillosilva-okta) [inline test/unit/spec/IDPDiscovery_spec.js]: description says `features.registration` is false, but here we are passing `null`. Should we change it to `false` or change the test description to match the test?

File: test/unit/spec/IDPDiscovery_spec.js [hunk 3]
```diff
@@ -1036,5 +1046,47 @@ function (_, $, Q, OktaAuth, LoginUtil, Okta, Util, AuthContainer, IDPDiscoveryF
...
+          expect(test.form.registrationContainer().length).toBe(0);
+        });
+      });
+      itp('show the registration button if settings.registration.enable is true', function () {
```
Comment (mauriciocastillosilva-okta) [inline test/unit/spec/IDPDiscovery_spec.js]: we should say ```if features.registration is enabled```, cause we don't really have a ```settings.registration.enable``` value

File: test/unit/spec/IDPDiscovery_spec.js (hunk 1 above)
Comment (haishengwu-okta) [reply test/unit/spec/IDPDiscovery_spec.js]: my understanding is RaaS and IDP discovery are sort of decoupled feature (thanks to Okta is always to be default idp) therefore we allow registration either primary auth or idp auth.

File: test/unit/spec/IDPDiscovery_spec.js [hunk 4]
```diff
@@ -1053,16 +1053,20 @@ function (_, $, Q, OktaAuth, LoginUtil, Okta, Util, AuthContainer, IDPDiscoveryF
           expect(test.form.registrationContainer().length).toBe(0);
         });
       });
+      itp('does not show the registration button if features.registration is not set', function () {
```
Comment (mauriciocastillosilva-okta) [inline test/unit/spec/IDPDiscovery_spec.js]: should probably say ```is undefined``` since the test above is for when ```it is not set```

**Pull request merged**